dynawo/sources/Models/CPP/Controls/Voltage/ModelMinMaxMean/test/TestModelMinMaxMean.cpp

```diff
@@ -0,0 +1,408 @@
+
+    unsigned nbY = nbVoltages;
+    unsigned nbF = 0;
+    unsigned nbZ = nbVoltages;
+    std::vector<propertyContinuousVar_t> yTypes(nbY, UNDEFINED_PROPERTY);
+    mmm->setBufferYType(&yTypes[0], 0);
+    ASSERT_NO_THROW(mmm->evalStaticYType());
+    ASSERT_EQ(mmm->sizeY(), nbY);
+    ASSERT_EQ(mmm->sizeF(), nbF);
+    ASSERT_EQ(mmm->sizeZ(), nbZ);
+    ASSERT_EQ(mmm->sizeG(), 0);
+    ASSERT_EQ(mmm->sizeMode(), 1);
+
+    mmm->evalStaticYType();
+    mmm->evalStaticFType();  // Does nothing here.
+    ASSERT_NO_THROW(mmm->initializeFromData(boost::shared_ptr<DataInterface>()));
+
+    std::vector<double> voltages(mmm->sizeY(), 0.);
+    // Binary variable for line connections
+    std::vector<double> z(mmm->sizeZ(), 0.);
+    bool* zConnected = new bool[mmm->sizeZ()];
+    unsigned int nbConnected = 2;
+    for (std::size_t i = 0; i < nbConnected; ++i) {
+        voltages[i] = static_cast<double>(i+1);
+        z[i] = 1.0;
+        zConnected[i] = true;
+    }
+    for (std::size_t i = nbConnected; i < nbVoltages; ++i) {
+        voltages[i] = static_cast<double>(i+1);
+        z[i] = -1.0;
+        zConnected[i] = true;
+    }
+    mmm->setBufferY(&voltages[0], nullptr, 0);
+    mmm->setBufferZ(&z[0], zConnected, 0);
+    mmm->evalCalculatedVars();
+    mmm->checkDataCoherence(0.);
+    mmm->initializeStaticData();
+    mmm->evalDynamicFType();
+    mmm->evalDynamicYType();
+
+    ASSERT_THROW_DYNAWO(mmm->evalCalculatedVarI(ModelMinMaxMean::nbCalculatedVars_), Error::MODELER, KeyError_t::UndefCalculatedVarI);
+}
+
+TEST(ModelsMinMaxMean, ModelsMinMaxMeanIndexesOfCalcVar) {
+    unsigned int nbVoltages = 25;
+    boost::shared_ptr<SubModel> mmm = initModelMinMaxMean(nbVoltages);
+    ASSERT_EQ(mmm->sizeY(), nbVoltages);
+
+    // Binary variable for line connections
+    std::vector<double> z(mmm->sizeZ(), 0.);
+    bool* zConnected = new bool[mmm->sizeZ()];
+
+    std::vector<boost::shared_ptr<Variable> > variables;
+    mmm->defineVariables(variables);
+
+    ASSERT_NO_THROW(mmm->initializeFromData(boost::shared_ptr<DataInterface>()));
+
+    std::vector<double> voltages(mmm->sizeY(), 0.);
+    for (std::size_t i = 0; i < nbVoltages; ++i) {
+        voltages[i] = static_cast<double>(i + 1);
+        z[i] = 1.0;
+        zConnected[i] = true;
+    }
+    mmm->setBufferY(&voltages[0], nullptr, 0);
+    mmm->setBufferZ(&z[0], zConnected, 0);
+    // Check the indexes needed for computing the min
+    std::vector<int> minIndexes;
+    mmm->getIndexesOfVariablesUsedForCalculatedVarI(DYN::ModelMinMaxMean::minValIdx_, minIndexes);
+
+    for (std::size_t i=0; i < nbVoltages; ++i) {
+        ASSERT_EQ(minIndexes[i], i);
+    }
+
+    // Check the indexes needed for computing the max
+    std::vector<int> maxIndexes;
+    mmm->getIndexesOfVariablesUsedForCalculatedVarI(DYN::ModelMinMaxMean::maxValIdx_, maxIndexes);
+
+    for (std::size_t i=0; i < nbVoltages; ++i) {
+        ASSERT_EQ(maxIndexes[i], i);
```

Comment:
I don't see a test where at least one input is not connected ?

Reply:
Very true! This has been handled, thanks!